### PR TITLE
Fix #1083 (Nil-pointer dereference)

### LINKED ISSUE #1083: Prevent nil pointer dereference
Contains*-Assertions panic when a `nil` instead of a string, slice, map etc. is injected.
Minimal example here:

```go
var myMap map[string]interface{} // empty
// try an assertion on nested keys
// which effectively leads to a passing of nil
require.Contains(t, myMap["some-key"], "expected-nested-key") // <-- panic
```

Assertions should be robust against nil-injection and gracefully fail instead of panicking.

### PR DESCRIPTION
**Pull request by mvrahden**
The fix gracefully fails the assertion when given `nil` as list value instead of panicking unresolved.

Fixes #1083

### CODE REVIEW

**Comment by mvrahden** [comment]
@ernesto-jimenez can anybody give a pass on this please? this is a trivial fix IMHO

**Comment by mvrahden** [comment]
As recommended in #1095 I'll kindly ask @boyan-soubachov to review this fix and provide feedback or merge.

File: assert/assertions.go [hunk 1]
```diff
@@ -721,7 +721,11 @@ func NotEqualValues(t TestingT, expected, actual interface{}, msgAndArgs ...inte
 func includeElement(list interface{}, element interface{}) (ok, found bool) {
```
Comment (boyan-soubachov) [inline assert/assertions.go:721]: Something seems wrong with the docstring of this function... Do you mind updating the function name in line 717?

**Review by boyan-soubachov — CHANGES_REQUESTED** [review]
Hmm. I think we need to migrate to GitHub actions since Travis killed support for open source a while ago if I recall correctly :/

**Comment by boyan-soubachov** [comment]
I've been a bit swamped with life for the past year or so (who knew 2020 and 2021 would be such shockers :D)

Thanks for your contribution. I'll first migrate the builds to GitHub actions and we can then rebase and merge your PR once the builds are done :)

File: assert/assertions.go (hunk 1 above)
Comment (mvrahden) [reply assert/assertions.go]: yes, they are not synced. I fixed that and fixed further linting issues in the `assert` package.

**Comment by mvrahden** [comment]
All good :) the last year was a bummer for many I guess 😅

I addressed your request with two additional commits.

There are further linting issues in other packages.
I might get back to them when I have some time and will make another PR for them.

**Comment by boyan-soubachov** [comment]
Hello, do you mind merging the latest master or rebasing this PR to trigger the CI builds

**Comment by mvrahden** [comment]
should be rebased now 👍

**Review by boyan-soubachov — APPROVED** [review]

**Comment by boyan-soubachov** [comment]
LGTM, thank you for your contribution!

**Comment by mvrahden** [comment]
happy to do so!

**Pull request merged**